src/server/server_state.h

```diff
@@ -267,6 +267,12 @@ class ServerState {  // public struct - to allow initialization.
     return slow_log_shard_;
   };
 
+  // Tries to returns as much RSS memory as possible to the OS.
+  // If data_heap_only is true, does it only for data_heap(), otherwise flushes the
+  // backing heap for this thread as well. If decommit_glibcmalloc is true, also
+  // decommits memory allocated by glibc malloc (globally).
+  void DecommitMemory(bool data_heap_only, bool decommit_glibcmalloc);
```

Comment:
Nit: I'm not a fan of accepting a `bool` for each path in the function. I personally would prefer to see either different functions (`DecommitDataHeap()`, `DecommitAll()`, ...) or an enum with meaningful names (`DecommitMemory(kDataAndMalloc)`).
Your call :)

Reply:
yeah, when I was studying a CS they told me I can use booleans as arguments.

Reply:
fixed